src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c

```diff
@@ -689,13 +689,14 @@ static int32_t GetCollationElementMask(UColAttributeValue strength)
     }
 }
 
-static int32_t inline SimpleAffix_Iterators(UCollationElements* pPatternIterator, UCollationElements* pSourceIterator, UColAttributeValue strength, int32_t forwardSearch)
+static int32_t inline SimpleAffix_Iterators(UCollationElements* pPatternIterator, UCollationElements* pSourceIterator, UColAttributeValue strength, int32_t forwardSearch, int32_t captureOffset, int32_t* pCapturedOffset)
 {
     assert(strength >= UCOL_SECONDARY);
```

Comment:
>     [](start = 0, length = 4)

could you please add assert when captureOffset is true, pCapturedOffset would be null (and possibly the reverse too)


maybe assert(captureOffset ^ pCapturedOffset == null)

Reply:
With your other suggestion of using only the pointer parameter (and dropping the bool), this is no longer necessary.

Reply:
that is right.